code/modules/reagents/reagent_containers.dm

```diff
@@ -149,6 +149,10 @@
 			MSG_VISUAL,
 			span_userdanger("You feel drenched!"),
 		)
+		if(iscarbon(target))
+			var/mob/living/carbon/drunkyard = target
+			if(drunkyard.has_trauma_type(/datum/brain_trauma/severe/split_personality/blackout) && prob(10))
+				drunkyard.cure_trauma_type(/datum/brain_trauma/severe/split_personality/blackout)
```

Comment:
You should signalize this instead

`SEND_SIGNAL(target, COMSIG_ATOM_SPLASHED, ...)` Then hook the signal in brain trauma apply